code/modules/mob/living/basic/lavaland/goldgrub/goldgrub.dm

```diff
@@ -0,0 +1,173 @@
+	AddElement(/datum/element/wall_smasher)
+	AddComponent(/datum/component/ai_listen_to_weather)
+	spit.Grant(src)
+	burrow.Grant(src)
+	ai_controller.set_blackboard_key(BB_SPIT_ABILITY, spit)
+	ai_controller.set_blackboard_key(BB_BURROW_ABILITY, burrow)
+
+/mob/living/basic/mining/goldgrub/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	. = ..()
+	if(!.)
+		return
+
+	if(!proximity_flag)
+		return
+
+	if(istype(attack_target, /obj/item/stack/ore))
+		consume_ore(attack_target)
+
+/mob/living/basic/mining/goldgrub/bullet_act(obj/projectile/bullet)
+	if(stat == DEAD)
+		return BULLET_ACT_FORCE_PIERCE
+	else
+		visible_message(span_danger("The [bullet.name] is repelled by [src]'s girth!"))
+		return BULLET_ACT_BLOCK
+
+/mob/living/basic/mining/goldgrub/proc/barf_contents()
+	visible_message(span_danger("[src] spits out its consumed ores!"))
+	playsound(src, 'sound/effects/splat.ogg', 50, TRUE)
+	for(var/obj/item/ore as anything in src)
+		ore.forceMove(loc)
+
+/mob/living/basic/mining/goldgrub/proc/generate_loot()
+	var/loot_amount = rand(1,3)
+	var/list/weight_lootdrops = list(
+		/obj/item/stack/ore/silver = 4,
+		/obj/item/stack/ore/gold = 3,
+		/obj/item/stack/ore/uranium = 3,
+		/obj/item/stack/ore/diamond = 1,
+	)
+	var/list/death_loot = list()
+	for(var/i in 1 to loot_amount)
+		death_loot += pick_weight(weight_lootdrops)
+	AddElement(/datum/element/death_drops, death_loot)
+
+/mob/living/basic/mining/goldgrub/death(gibbed)
+	barf_contents()
+	return ..()
+
+/mob/living/basic/mining/goldgrub/proc/make_tameable()
+	AddComponent(\
+		/datum/component/tameable,\
+		food_types = list(/obj/item/stack/ore),\
+		tame_chance = 25,\
+		bonus_tame_chance = 5,\
+		after_tame = CALLBACK(src, PROC_REF(tame_grub)),\
+	)
+
+/mob/living/basic/mining/goldgrub/proc/tame_grub()
+	new /obj/effect/temp_visual/heart(src.loc)
+	AddElement(/datum/element/ridable, /datum/component/riding/creature/goldgrub)
+	AddComponent(/datum/component/obeys_commands, pet_commands)
+
+/mob/living/basic/mining/goldgrub/proc/make_egg_layer()
+	AddComponent(\
+		/datum/component/egg_layer,\
+		/obj/item/food/egg/green/grub_egg,\
+		list(/obj/item/stack/ore/bluespace_crystal),\
+		lay_messages = EGG_LAYING_MESSAGES,\
+		eggs_left = 0,\
+		eggs_added_from_eating = 1,\
+		max_eggs_held = 1,\
+		egg_laid_callback = CALLBACK(src, PROC_REF(lay_grub_egg)),\
+	)
+
+/mob/living/basic/mining/goldgrub/proc/lay_grub_egg(obj/item/grub_egg)
+	if(prob(1))
+		return
+	grub_egg.AddComponent(\
+		/datum/component/fertile_egg,\
+		embryo_type = /mob/living/basic/mining/goldgrub/baby,\
+		minimum_growth_rate = 1,\
+		maximum_growth_rate = 2,\
+		total_growth_required = 100,\
+		current_growth = 0,\
+		location_allowlist = typecacheof(list(/turf)),\
+	)
+
+/mob/living/basic/mining/goldgrub/proc/consume_ore(obj/item/target_ore)
+	playsound(src.loc,'sound/items/eatfood.ogg', rand(10,50), TRUE)
+	target_ore.forceMove(src)
+
+/mob/living/basic/mining/goldgrub/baby
```

Comment:
Consider overriding `generate_loot` for these to not spawn with ores in them (since they're babby)